graphql_compiler/compiler/ir_lowering_match/__init__.py

```diff
@@ -103,4 +104,4 @@ def lower_ir(ir_blocks, location_types, type_equivalence_hints=None):
     compound_match_query = lower_context_field_expressions(
         compound_match_query)
 
-    return compound_match_query
+    return lower_sub_queries(compound_match_query)
```

Comment:
Please preserve the structure and style of the function -- more passes will be added, and people shouldn't have to modify the return line:
```
compound_match_query = lower_sub_queries(compound_match_query)

return compound_match_query
```